cebra/data/assets.py

```diff
@@ -0,0 +1,129 @@
+#
+# (c) All rights reserved. ECOLE POLYTECHNIQUE FÉDÉRALE DE LAUSANNE,
+# Switzerland, Laboratory of Prof. Mackenzie W. Mathis (UPMWMATHIS) and
+# original authors: Steffen Schneider, Jin H Lee, Mackenzie W Mathis. 2023.
+#
+# Source code:
+# https://github.com/AdaptiveMotorControlLab/CEBRA
+#
+# Please see LICENSE.md for the full license document:
+# https://github.com/AdaptiveMotorControlLab/CEBRA/LICENSE.md
+#
+
+import hashlib
+import os
+import re
+import warnings
+from typing import Optional
+
+import requests
+import tqdm
+
+_MAX_RETRY_COUNT = 2
+
+
+def download_file_with_progress_bar(url: str,
+                                    expected_checksum: str,
+                                    location: str,
+                                    file_name: str,
+                                    retry_count: int = 0) -> Optional[str]:
+    """
+    Downloads a file from the given URL with a progress bar and performs checksum verification.
+
+    Args:
+        url: The URL to download the file from.
+        expected_checksum: The expected checksum value of the downloaded file.
+        location: The directory where the file will be saved.
+        file_name: The name of the file.
+        retry_count: The number of retry attempts (default: 0).
+
+    Returns:
+        The path of the downloaded file if the download is successful, None otherwise.
+
+    Raises:
+        RuntimeError: If the maximum retry count is exceeded.
+    """
```

Comment:
```suggestion
    """Download a file from the given URL.
    
    During download, progress is reported using a progress bar. The downloaded
    file's checksum is compared to the provided ``expected_checksum``.

    Args:
        url: The URL to download the file from.
        expected_checksum: The expected checksum value of the downloaded file.
        location: The directory where the file will be saved.
        file_name: The name of the file.
        retry_count: The number of retry attempts (default: 0).

    Returns:
        The path of the downloaded file if the download is successful, None otherwise.

    Raises:
        RuntimeError: If the maximum ``retry count`` is exceeded.
    """
```